scripts/extract-package-code.R

```diff
@@ -126,3 +138,101 @@ extract_package_vignettes <- function(pkg, pkg_dir, output_dir) {
 
     vignettes
 }
+
+
+path_sanitized <- function(path) {
+    path_abs(path_tidy(path), getwd())
+}
+
+parse_command_line_arguments <- function() {
+
+    option_list <- list(
+        make_option(c("-r", "--r-dyntrace"), action="store", type="character",
+                    default="../../R-dyntrace/bin/R", help="R-dyntrace interpreter.",
+                    metavar="r-dyntrace"),
+        make_option(c("-v", "--verbose"), action="store_true", default=FALSE,
+                    help="Flag to enable verbose mode.", metavar="verbose"),
+        make_option(c("--corpus-dirpath"), action="store", type="character",
+                    help="Directory for storing vignettes.", metavar="corpus-dirpath"),
+        make_option(c("--raw-analysis-dirpath"), action="store", type="character",
+                    help="Directory for tracer generated data.", metavar="raw-analysis-dirpath"),
+        make_option(c("--disable-eval-expression-analysis"), action="store_true", default=FALSE,
+                    help="Flag to disable eval expression analysis.", metavar="eval-expression-analysis")
+    )
+
+    args <- parse_args(OptionParser(option_list = option_list),
+                       positional_arguments=TRUE)
+
+    list(package = args$args[1],
+         verbose = args$options$verbose,
+         r_dyntrace = path_sanitized(args$options$`r-dyntrace`),
+         corpus_dirpath = path_sanitized(args$options$`corpus-dirpath`),
+         raw_analysis_dirpath = path_sanitized(args$options$`raw-analysis-dirpath`),
+         analysis_flags =
+             list(eval_expression_analysis = !args$options$`disable-eval-expression-analysis`))
+}
+
+
+copy <- function(settings) {
+    extract_package_code(settings$package, types="all",
+                         output_dir = path(settings$corpus_dirpath, settings$package));
+}
+
+wrap <- function(package_code, settings) {
+
+    indent <- function(lines, spaces=4) {
```

Comment:
Wouldn't it be easier to pass the result to `formatR`?

Reply:
I don't want to radically reformat user code. I just want to indent the user code by 4 spaces so that it appears indented in the `dyntrace_evals` block. Plus, calling `formatR` make make this process very slow.